packages/react-client/src/hooks/devices.ts

```diff
@@ -1,33 +1,98 @@
 import { useFishjamContext } from "./fishjamContext";
 import type { Device, AudioDevice } from "../types";
+import { useVideoDeviceManager } from "./deviceManagers/videoDeviceManager";
+import { useAudioDeviceManager } from "./deviceManagers/audioDeviceManager";
+import { useCallback, useEffect, useRef } from "react";
+import { getAvailableMedia, getCorrectedResult } from "../mediaInitializer";
 
 export function useCamera(): Device {
   const {
-    state,
     videoTrackManager: { currentTrack, ...trackManager },
   } = useFishjamContext();
 
+  const { deviceState } = useVideoDeviceManager();
+
   const streamedTrackId = currentTrack?.trackId ?? null;
   const streamedTrack = currentTrack?.track ?? null;
   const stream = currentTrack?.stream ?? null;
-  const devices = state.devices.camera.devices ?? [];
-  const activeDevice = state.devices.camera.deviceInfo;
+  const devices = deviceState.devices ?? [];
+  const activeDevice = deviceState.media?.deviceInfo ?? null;
 
   return { ...trackManager, streamedTrack, streamedTrackId, stream, devices, activeDevice };
 }
 
 export function useMicrophone(): AudioDevice {
   const {
-    state,
     audioTrackManager: { currentTrack, ...trackManager },
   } = useFishjamContext();
 
+  const { deviceState } = useAudioDeviceManager();
+
   const isAudioPlaying = currentTrack?.vadStatus === "speech";
   const streamedTrackId = currentTrack?.trackId ?? null;
   const streamedTrack = currentTrack?.track ?? null;
   const stream = currentTrack?.stream ?? null;
-  const devices = state.devices.microphone.devices ?? [];
-  const activeDevice = state.devices.microphone.deviceInfo;
+  const devices = deviceState.devices ?? [];
+  const activeDevice = deviceState.media?.deviceInfo ?? null;
 
   return { ...trackManager, streamedTrack, streamedTrackId, isAudioPlaying, stream, devices, activeDevice };
 }
+
+type InitializeDevicesProps = { autoInitialize?: boolean };
+
+export const useInitializeDevices = (props?: InitializeDevicesProps) => {
```

Comment:
**Future improvement idea:** I think it would be beneficial for the user to be allowed to request access to only the camera or the microphone. By default, it should start both, as you have implemented.

Reply:
https://linear.app/swmansion/issue/FCE-457/allow-user-to-only-initialise-audio-or-video